Crashing bug

Error: signal 11:
proxysql(_Z13crash_handleri+0x20)[0x4a5fe0]
/lib/x86_64-linux-gnu/libc.so.6(+0x36150)[0x7f19470ac150]
/lib/x86_64-linux-gnu/libc.so.6(+0x162231)[0x7f19471d8231]
proxysql(_ZN14ProxySQL_Admin25stats___mysql_processlistEv+0x8d)[0x4d7a3d]
proxysql(_ZN14ProxySQL_Admin24GenericRefreshStatisticsEPKcjb+0x22a)[0x4d94ea]
proxysql(_Z21admin_session_handlerP13MySQL_SessionP14ProxySQL_AdminP10_PtrSize_t+0xdd)[0x4dfaed]
proxysql(_ZN13MySQL_Session7handlerEv+0x1477)[0x4c8757]
proxysql(_Z11child_mysqlPv+0x365)[0x4d8615]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x7e9a)[0x7f1948692e9a]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x6d)[0x7f194716938d]
2015-09-29 20:31:41 [ERROR] ProxySQL exited with code 1 . Restarting!
2015-09-29 20:31:41 [INFO] Angel process is waiting 1 seconds before starting a new ProxySQL process
2015-09-29 20:31:42 [INFO] Angel process started ProxySQL process 11142

--- a/lib/ProxySQL_Admin.cpp
+++ b/lib/ProxySQL_Admin.cpp
@@ -2495,10 +2495,26 @@ void ProxySQL_Admin::stats___mysql_processlist() {
 		SQLite3_row *r=*it;
 		int arg_len=0;
 		for (int i=0; i<14; i++) {
-			arg_len+=strlen(r->fields[i]);
+			if (r->fields[i])
+				arg_len+=strlen(r->fields[i]);
 		}
 		char *query=(char *)malloc(strlen(a)+arg_len+32);
-		sprintf(query,a,r->fields[0],r->fields[1],r->fields[2],r->fields[3],r->fields[4],r->fields[5],r->fields[6],r->fields[7],r->fields[8],r->fields[9],r->fields[10],r->fields[11],r->fields[12],r->fields[13]);
+		sprintf(query,a,
+			(r->fields[0] ? r->fields[0] : ""),
+			(r->fields[1] ? r->fields[1] : ""),
+			(r->fields[2] ? r->fields[2] : ""),
+			(r->fields[3] ? r->fields[3] : ""),
+			(r->fields[4] ? r->fields[4] : ""),
+			(r->fields[5] ? r->fields[5] : ""),
+			(r->fields[6] ? r->fields[6] : ""),
+			(r->fields[7] ? r->fields[7] : ""),
+			(r->fields[8] ? r->fields[8] : ""),
+			(r->fields[9] ? r->fields[9] : ""),
+			(r->fields[10] ? r->fields[10] : ""),
+			(r->fields[11] ? r->fields[11] : ""),
+			(r->fields[12] ? r->fields[12] : ""),
+			(r->fields[13] ? r->fields[13] : "")
+		);
 		statsdb->execute(query);
 		free(query);
 	}

